testing/upgradetest/versions_test.go

```diff
@@ -5,88 +5,128 @@
 		"7.17",
-		"8.7",
-		"8.8",
 		"8.9",
-		"8.10-SNAPSHOT",
 		"8.10",
-		"8.11-SNAPSHOT",
 		"8.11",
 		"8.12-SNAPSHOT",
+		"8.12",
+		"8.13-SNAPSHOT",
+		"8.13",
+		"8.14-SNAPSHOT",
 	},
 	Manifests: tools.Manifests{
-		LastUpdateTime:         "Tue, 10 Oct 2023 19:20:17 UTC",
-		SecondsSinceLastUpdate: 278,
+		LastUpdateTime:         "Fri, 23 Feb 2024 11:25:33 UTC",
+		SecondsSinceLastUpdate: 164,
 	},
 }
 
-// Tests that GetUpgradableVersions behaves correctly during the feature freeze period
-// where the both main and the previous minor release branch versions are unreleased.
-// Regression test for the problem described in https://github.com/elastic/elastic-agent/pull/3563#issuecomment-1756007790.
-func TestGetUpgradableVersionsAfterFeatureFreeze(t *testing.T) {
+func TestFetchUpgradableVersionsAfterFeatureFreeze(t *testing.T) {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
-	// Start from 8.12.0 assuming the 8.11.0 feature freeze has just happened.
-	// The 8.11.0 release is upgradable because the first 8.11.0 build candidate exists,
-	// but it is only available from staging.elastic.co which is not a binary download
-	// source that is supported by default.
-	currentVersion := "8.12.0"
-
-	// Since the 8.11.0 BC at staging.elastic.co isn't available to the agent by default,
-	// getUpgradableVersions should return 8.11.0-SNAPSHOT as the previous minor so an
-	// upgrade can proceed.
 	expectedUpgradableVersions := []string{
-		"8.11.0-SNAPSHOT", "8.10.3", "8.10.2", "7.17.14", "7.17.13",
+		"8.13.0-SNAPSHOT",
+		"8.12.2",
+		"8.12.2-SNAPSHOT",
+		"8.12.1",
+		"8.12.0",
+		"8.11.4",
+		"7.17.18",
 	}
 
-	// Get several of the previous versions to ensure snapshot selection works correctly.
-	versions, err := getUpgradableVersions(ctx, &versionListAfter8_11FeatureFreeze, currentVersion, 3, 2)
+	versionBytes, err := json.Marshal(versionsResponse)
 	require.NoError(t, err)
-	require.NotEmpty(t, versions)
 
+	server := httptest.NewServer(http.HandlerFunc(func(resp http.ResponseWriter, req *http.Request) {
+		resp.WriteHeader(http.StatusOK)
+		_, err := io.Copy(resp, bytes.NewBuffer(versionBytes))
+		assert.NoError(t, err)
+	}))
+	defer server.Close()
+	aac := tools.NewArtifactAPIClient(tools.WithUrl(server.URL))
+
+	reqs := VersionRequirements{
+		UpgradeToVersion: "8.13.0", // to test that 8.14 is not returned
+		CurrentMajors:    3,        // should return 8.12.2, 8.12.1, 8.12.0
+		PreviousMajors:   3,        // should return 7.17.18
+		PreviousMinors:   2,        // should return 8.12.2, 8.11.4
+		RecentSnapshots:  2,        // should return 8.13.0-SNAPSHOT, 8.12.2-SNAPSHOT
+	}
+	versions, err := FetchUpgradableVersions(ctx, aac, reqs)
+	require.NoError(t, err)
 	t.Logf("exp: %s", expectedUpgradableVersions)
 	t.Logf("act: %s", versions)
-	for i, exp := range expectedUpgradableVersions {
-		require.Equal(t, exp, versions[i].String())
-	}
+	require.Equal(t, expectedUpgradableVersions, versions)
```

Comment:
[Suggestion]
This is not a case for `require`. It's just a test that if it fails and the code would proceed it would not lead to unexpected behaviour or a panic

```suggestion
	assert.Equal(t, expectedUpgradableVersions, versions)
```